src/cinema/presentation/CinemaController.java

```diff
@@ -0,0 +1,61 @@
+package cinema.presentation;
+
+import cinema.businesslayer.CinemaService;
+import cinema.domainmodel.Seat;
+import cinema.domainmodel.Token;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+public class CinemaController {
+
+    @Autowired
+    CinemaService cinemaService;
+
+    @GetMapping("/seats")
+    public SeatsDto getSeats() {
+        return cinemaService.getSeats();
+    }
+
+    @PostMapping("/purchase")
+    public TicketDto purchaseTicket(@RequestBody Seat seat) {
+        if (!cinemaService.checkIfRowColumnValid(seat.getRow(), seat.getColumn())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The number of a row or a column is out of bounds!");
+        } else {
+            if (cinemaService.checkIfTicketPurchased(seat.getRow(), seat.getColumn())) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The ticket has been already purchased!");
+            } else {
+                return cinemaService.purchaseSeat(seat.getRow(), seat.getColumn());
+            }
+        }
```

Comment:
That pice of code looks a bit overcomplicated in terms of Cyclomatic Complexity
https://www.geeksforgeeks.org/cyclomatic-complexity/

There is a better way of writing such type of code using guards syntax:
```java
        if (!cinemaService.checkIfRowColumnValid(seat.getRow(), seat.getColumn())) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The number of a row or a column is out of bounds!");
        }
        if (cinemaService.checkIfTicketPurchased(seat.getRow(), seat.getColumn())) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The ticket has been already purchased!");
        }
        return cinemaService.purchaseSeat(seat.getRow(), seat.getColumn());
```
It reduces the complexity and more readable.

Reply:
fixed